jax/experimental/array.py

```diff
@@ -95,7 +95,7 @@ def _reconstruct_array(fun, args, arr_state, aval_state):
 
 
 @pxla.use_cpp_class(xc.Array if xc._version >= 92 else None)
-class Array:
+class Array(basearray.Array):
```

Comment:
Note that if the C++ class is in use (i.e., you have a new enough `jaxlib`) then subtyping won't hold for `Array` either.

Reply:
Ah, right, so maybe for now we do need `basearray.Array.register(Array)`, which will take care of runtime instance checks. At static type check time, though, I don't think mypy/pytype is smart enough to figure out that the decorator has replaced the class implementation, so I believe it will still work as long as this is the method used to fall-back. What do you think?

Reply:
Maybe just do the register below even for Python?

Reply:
I've done it conditionally based on the jaxlib version, let me know if that looks OK.

Reply:
No, if we remove the inheritance then mypy/pytype will no longer recognize `Array` as a subclass of `jax.Array`

Reply:
Ohh I see. Okay, LGTM.